### PR TITLE
Disallow CORS on unsupported origins

### PR DESCRIPTION
**Pull request by nicl**
It's unclear why this was supported before, but allowing CORS on all origins doesn't seem like a sensible default.

Calling code can still provide a fallback origin to support if needed.

It is hard to know up front if this will break things. We have a whitelist of origins in our config with a list of seemingly sensible values, but it's possible there are other origins we need to add too. However, our Kibana logs for Frontend show that in the last 24 hours 100% of requests with an origin value were for `https://www.theguardian.com`, which suggests things should be okay.

## What does this change?

Remove CORS support for non-whitelisted origins.

## What is the value of this and can you measure success?

Reduces future risk of a cors security issue.

### CODE REVIEW

File: common/app/model/Cors.scala [hunk 1]
```diff
@@ -8,20 +8,25 @@ object Cors extends Results with implicits.Requests {
...
-        result.withHeaders(headers: _*)
+        allowedOrigin match {
+          case Some(origin) =>
+            val headers = allowedMethods.map("Access-Control-Allow-Methods" -> _).toList ++ List(
+              "Access-Control-Allow-Origin" -> origin,
+              "Access-Control-Allow-Headers" -> responseHeaders,
+              "Access-Control-Allow-Credentials" -> "true")
+
+            result.withHeaders(headers: _*)
+          case None => Forbidden("Unsupported CORS origin")
```
Comment (rahilb) [inline common/app/model/Cors.scala]: I think just not returning the `Access-Control-Allow-Origin` header will produce the same behaviour in browsers, though returning a 403 might be better as we can see in the logs how often it happens?

Comment (nicl) [reply common/app/model/Cors.scala]: Ah maybe that is better. I wasn't sure what was the 'right' approach to error handling here.

Comment (rahilb) [reply common/app/model/Cors.scala]: yep I'm not too sure either: the 403 is nice as we can easily see if anyone is affected.

**Review by rahilb — APPROVED** [review]
LGTM

**Pull request merged**